utility/src/main/java/org/oppia/android/util/profile/ProfileNameValidator.kt

```diff
@@ -0,0 +1,55 @@
+package org.oppia.android.util.profile
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Utility to validate that profile names are correctly formatted. */
+@Singleton
+class ProfileNameValidator @Inject constructor() {
+  private val noRepeatedAllowedSymbolsRegex by lazy { Regex("[.'-]{2}") }
+
+  /**
+   * Validates a profile name to ensure that it isn't confusingly formatted
+   * or contains invalid characters.
```

Comment:
Two nits:
1. Please make sure documentation lines take up as much horizontal space as possible before line-wrapping (i.e. up to 100 characters which is our line limit).
2. Perhaps add some clarifying details about what 'invalid' means (e.g. non-letter characters, per the Unicode standard for letter categories, plus apostrophes and dashes).

Reply:
done